salt/utils/win_runas.py

```diff
@@ -44,422 +39,135 @@ def __virtual__():
-                                          win32con.HANDLE_FLAG_INHERIT)
-        if inherit_write:
-            kernel32.SetHandleInformation(write, win32con.HANDLE_FLAG_INHERIT,
-                                          win32con.HANDLE_FLAG_INHERIT)
-        return read.value, write.value
-
-    def CreateProcessWithLogonW(username=None,
-                                domain=None,
-                                password=None,
-                                logonflags=0,
-                                applicationname=None,
-                                commandline=None,
-                                creationflags=0,
-                                environment=None,
-                                currentdirectory=None,
-                                startupinfo=None):
-        creationflags |= win32con.CREATE_UNICODE_ENVIRONMENT
-        if commandline is not None:
-            commandline = ctypes.create_unicode_buffer(commandline)
-        if startupinfo is None:
-            startupinfo = STARTUPINFO()
-        process_info = PROCESS_INFORMATION()
-        advapi32.CreateProcessWithLogonW(username,
-                                         domain,
-                                         password,
-                                         logonflags,
-                                         applicationname,
-                                         commandline,
-                                         creationflags,
-                                         environment,
-                                         currentdirectory,
-                                         ctypes.byref(startupinfo),
-                                         ctypes.byref(process_info))
-        return process_info
-
-
-def make_inheritable(token):
-    return win32api.DuplicateHandle(win32api.GetCurrentProcess(),
-                                    token,
-                                    win32api.GetCurrentProcess(),
-                                    0,
-                                    1,
-                                    win32con.DUPLICATE_SAME_ACCESS)
-
-
-def runas_system(cmd, username, password):
-    # This only works as system, when salt is running as a service for example
-
-    # Check for a domain
+def split_username(username):
+    # TODO: Is there a windows api for this?
     domain = '.'
     if '@' in username:
         username, domain = username.split('@')
     if '\\' in username:
         domain, username = username.split('\\')
+    return username, domain
+
+
+def runas(cmdLine, username, password=None, cwd=None, elevated=True):
+
+    impersonation_token = salt.win.impersonate_sid(
+        salt.win.SYSTEM_SID,
+        session_id=0,
+        privs=['SeTcbPrivilege'],
+    )
+
+    username, domain = split_username(username)
+    sid, domain, sidType = win32security.LookupAccountName(domain, username)
+    if domain == 'NT AUTHORITY':
+        log.warn("Logon system account: %s", username)
```

Comment:
Why warning?
Also, switch to `log.warning`, `log.warn` is deprecated in Py3.

Reply:
I agree warning is overkill. Should these 'logon user' log statements go away completely?

Reply:
If you think these will make sense when trying to debug an issue, leave then, but perhaps, at debug log level? Info?

Reply:
Windows creates an event log for the logons so I've just removed these log statements.